output/github.go

```diff
@@ -0,0 +1,94 @@
+package output
+
+import (
+	"fmt"
+	"io"
+)
+
+// GitHub represents an Outputter that outputs
+// results in GitHub workflow format.
+type GitHub struct {
+	writer io.Writer
+}
+
+// NewGitHub creates a new GitHub with the given writer.
+func NewGitHub(w io.Writer) *GitHub {
+	github := GitHub{
+		writer: w,
+	}
+
+	return &github
+}
+
+// Output outputs the results.
+func (t *GitHub) Output(checkResults []CheckResult) error {
+	var totalFailures int
+	var totalExceptions int
+	var totalWarnings int
+	var totalSuccesses int
+	var totalSkipped int
+	for _, result := range checkResults {
+		totalPolicies := result.Successes + len(result.Failures) + len(result.Warnings) + len(result.Exceptions) + len(result.Skipped)
+
+		fmt.Fprintf(t.writer, "::group::Testing '%v' against %v policies in namespace '%v'\n", result.FileName, totalPolicies, result.Namespace)
+		for _, failure := range result.Failures {
+			fmt.Fprintf(t.writer, "::error file=%v::%v\n", result.FileName, failure.Message)
+		}
+
+		for _, warning := range result.Warnings {
+			fmt.Fprintf(t.writer, "::warning file=%v::%v\n", result.FileName, warning.Message)
+		}
+
+		for _, exception := range result.Exceptions {
+			fmt.Fprintf(t.writer, "exception file=%v::%v\n", result.FileName, exception.Message)
+		}
+
+		for _, skipped := range result.Skipped {
+			fmt.Fprintf(t.writer, "skipped file=%v::%v\n", result.FileName, skipped.Message)
+		}
+
+		if result.Successes > 0 {
+			fmt.Fprintf(t.writer, "success file=%v %v\n", result.FileName, result.Successes)
```

Comment:
is this meant to not have the :: like the others?

Reply:
worth adding a test with a success case in perhaps

Reply:
Sadly only errors and warning are annotated in GitHub PR's and no success is set. If there are no errors, the check will return 'green'. The '::' is something that GitHub picks up.

Reply:
I have removed all the double colons where the output is not GitHub flavored (exception/skipped/success).